sdk/core/azure-core-serializer-json-gson/src/main/java/com/azure/core/serializer/json/gson/GsonJsonSerializer.java

```diff
@@ -65,4 +71,26 @@
     public <S extends OutputStream> Mono<S> serializeAsync(S stream, Object value) {
         return Mono.fromCallable(() -> serialize(stream, value));
     }
+
+    @Override
+    public String convertMemberName(Member member) {
+        if (Modifier.isTransient(member.getModifiers())) {
+            return null;
+        }
+        if (member instanceof Field) {
+            Field f = (Field) member;
+            if (gson.excluder().excludeField(f, true)) {
+                return null;
+            }
+            if (f.isAnnotationPresent(SerializedName.class)) {
+                return f.getDeclaredAnnotation(SerializedName.class).value();
+            }
+            return member.getName();
+        }
+        if (member instanceof Method) {
```

Comment:
Did we ever determine whether GSON uses `Methods` during serialization?

Reply:
From what we determined yesterday, we need to return null here. Will fix it.
We only deal with the member which can be renamed through annotation.